tests/integration2/tunnel/tunnel_test.go

```diff
@@ -0,0 +1,351 @@
+package tunnel
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/log"
+	apps2 "istio.io/istio/pkg/test/framework/runtime/components/apps"
+
+	v1 "k8s.io/api/core/v1"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/api/components"
+	"istio.io/istio/pkg/test/framework/api/descriptors"
+	"istio.io/istio/pkg/test/framework/api/ids"
+	"istio.io/istio/pkg/test/framework/api/lifecycle"
+	"istio.io/istio/pkg/test/framework/runtime/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/tmpl"
+)
+
+const (
+	clientSideEgressConfig = `
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  labels:
+    service: client
+  name: istio-egressgateway-client
+spec:
+  selector:
+    istio: egressgateway
+  servers:
+  - hosts:
+    - {{ .sidecarSNI }}
+    port:
+      name: tcp-port-443
+      number: 443
+      protocol: TLS
+    tls:
+      caCertificates: /etc/certs/root-cert.pem
+      mode: MUTUAL
+      privateKey: /etc/certs/key.pem
+      serverCertificate: /etc/certs/cert-chain.pem
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: egress-gateway-client
+spec:
+  gateways:
+  - istio-egressgateway-client
+  hosts:
+  - {{ .sidecarSNI }}
+  tcp:
+  - match:
+    - gateways:
+      - istio-egressgateway-client
+      port: 443
+    route:
+    - destination:
+        host: {{ .ingressDNS }}
+        port:
+          number: 443
+        subset: client-2
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: egressgateway-client
+  namespace: istio-system
+spec:
+  host: {{ .ingressDNS }}
+  exportTo: [ "." ]
+  subsets:
+  - name: client-2
+    trafficPolicy:
+      portLevelSettings:
+      - port:
+          number: 443
+        tls:
+          caCertificates: /etc/istio/tunnel-certs/ca.crt
+          clientCertificate: /etc/istio/tunnel-certs/client.crt
+          mode: MUTUAL
+          privateKey: /etc/istio/tunnel-certs/client.key
+          sni: {{ .ingressDNS }}
+          subjectAltNames:
+          - {{ .ingressDNS }}
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
```

Comment:
Would it make sense to just always create a `ServiceEntry` as part of creating an ingress? I suspect this won't be the only test that tries sending traffic to the ingress.